diff -ur xfce4-genmon-plugin-3.2-orig/configure.ac xfce4-genmon-plugin-3.2/configure.ac
--- xfce4-genmon-plugin-3.2-orig/configure.ac	2011-03-01 11:20:52.947067638 +0100
+++ xfce4-genmon-plugin-3.2/configure.ac	2011-03-01 13:00:58.485889236 +0100
@@ -32,7 +32,7 @@
 
 dnl configure the panel plugin
 XDT_CHECK_PACKAGE([LIBXFCE4PANEL], [libxfce4panel-1.0], [4.3.22])
-XDT_CHECK_PACKAGE([LIBXFCEGUI4], [libxfcegui4-1.0], [4.3.22])
+XDT_CHECK_PACKAGE([LIBXFCE4UI], [libxfce4ui-1], [4.8.0])
 
 dnl Translations
 XDT_I18N([ca cs en_GB eu fr pl pt_PT sq])
diff -ur xfce4-genmon-plugin-3.2-orig/panel-plugin/config_gui.c xfce4-genmon-plugin-3.2/panel-plugin/config_gui.c
--- xfce4-genmon-plugin-3.2-orig/panel-plugin/config_gui.c	2007-11-23 16:32:40.000000000 +0100
+++ xfce4-genmon-plugin-3.2/panel-plugin/config_gui.c	2011-03-01 13:00:58.486889236 +0100
@@ -28,8 +28,8 @@
 #include <gtk/gtk.h>
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/dialogs.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
+#include <libxfce4ui/libxfce4ui.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 
 #define COPYVAL(var, field)    ((var)->field = field)
diff -ur xfce4-genmon-plugin-3.2-orig/panel-plugin/config_gui.h xfce4-genmon-plugin-3.2/panel-plugin/config_gui.h
--- xfce4-genmon-plugin-3.2-orig/panel-plugin/config_gui.h	2007-11-23 16:32:40.000000000 +0100
+++ xfce4-genmon-plugin-3.2/panel-plugin/config_gui.h	2011-03-01 13:00:58.486889236 +0100
@@ -29,8 +29,8 @@
 #include <gtk/gtk.h>
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/dialogs.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
+#include <libxfce4ui/libxfce4ui.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 
 typedef struct gui_t {
diff -ur xfce4-genmon-plugin-3.2-orig/panel-plugin/main.c xfce4-genmon-plugin-3.2/panel-plugin/main.c
--- xfce4-genmon-plugin-3.2-orig/panel-plugin/main.c	2008-03-15 23:38:18.000000000 +0100
+++ xfce4-genmon-plugin-3.2/panel-plugin/main.c	2011-03-01 17:15:23.747567404 +0100
@@ -31,9 +31,8 @@
 #include <gtk/gtk.h>
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/dialogs.h>
-#include <libxfce4panel/xfce-panel-plugin.h>
-#include <libxfce4panel/xfce-panel-convenience.h>
+#include <libxfce4ui/libxfce4ui.h>
+#include <libxfce4panel/libxfce4panel.h>
 
 #include <stdlib.h>
 #include <unistd.h>
@@ -584,7 +583,7 @@
 static void About (Widget_t w, void *unused)
 /* Called back when the About button in clicked */
 {
-    xfce_info (_("%s %s - Generic Monitor\n"
+    xfce_dialog_show_info(NULL, NULL, _("%s %s - Generic Monitor\n"
         "Cyclically spawns a script/program, captures its output "
         "and displays the resulting string in the panel\n\n"
         "(c) 2004 Roger Seguin <roger_seguin@msn.com>\n"
@@ -638,7 +637,7 @@
 /* Plugin API */
 /* Create/pop up the configuration/options GUI */
 {
-    GtkWidget *dlg, *header, *vbox;
+    GtkWidget *dlg, *vbox;
     struct param_t *poConf = &(poPlugin->oConf.oParam);
     struct gui_t   *poGUI = &(poPlugin->oConf.oGUI);
     const char     *pcFont = poConf->acFont;
@@ -647,25 +646,20 @@
 
     xfce_panel_plugin_block_menu (plugin);
 
-    dlg = gtk_dialog_new_with_buttons (_("Configuration"),
+    dlg = xfce_titled_dialog_new_with_buttons (_("Configuration"),
         GTK_WINDOW (gtk_widget_get_toplevel (GTK_WIDGET (plugin))),
         GTK_DIALOG_DESTROY_WITH_PARENT |
         GTK_DIALOG_NO_SEPARATOR,
         GTK_STOCK_CLOSE, GTK_RESPONSE_OK,
         NULL);
 
+    xfce_titled_dialog_set_subtitle (XFCE_TITLED_DIALOG(dlg), _("Generic Monitor"));
+
     g_signal_connect (dlg, "response", G_CALLBACK (genmon_dialog_response),
         poPlugin);
 
     gtk_container_set_border_width (GTK_CONTAINER (dlg), 2);
 
-    header = xfce_create_header (NULL, _("Generic Monitor"));
-    gtk_widget_set_size_request (GTK_BIN (header)->child, -1, 32);
-    gtk_container_set_border_width (GTK_CONTAINER (header), BORDER - 2);
-    gtk_widget_show (header);
-    gtk_box_pack_start (GTK_BOX (GTK_DIALOG (dlg)->vbox), header,
-        FALSE, TRUE, 0);
-
     vbox = gtk_vbox_new(FALSE, BORDER);
     gtk_container_set_border_width (GTK_CONTAINER (vbox), BORDER - 2);
     gtk_widget_show(vbox);
diff -ur xfce4-genmon-plugin-3.2-orig/panel-plugin/Makefile.am xfce4-genmon-plugin-3.2/panel-plugin/Makefile.am
--- xfce4-genmon-plugin-3.2-orig/panel-plugin/Makefile.am	2011-03-01 11:20:52.946067639 +0100
+++ xfce4-genmon-plugin-3.2/panel-plugin/Makefile.am	2011-03-01 13:00:58.490889236 +0100
@@ -4,11 +4,11 @@
 xfce4_genmon_plugin_CFLAGS =					\
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\"			\
 	@LIBXFCE4PANEL_CFLAGS@					\
-	@LIBXFCEGUI4_CFLAGS@
+	@LIBXFCE4UI_CFLAGS@
 
 xfce4_genmon_plugin_LDADD =					\
 	@LIBXFCE4PANEL_LIBS@					\
-	@LIBXFCEGUI4_LIBS@
+	@LIBXFCE4UI_LIBS@
 
 xfce4_genmon_plugin_SOURCES =	\
 	main.c			\
